tools/rosbag/src/recorder.cpp

```diff
@@ -347,7 +347,11 @@ void Recorder::doQueue(const ros::MessageEvent<topic_tools::ShapeShifter const>&
                 ros::M_string::const_iterator it2 = out.connection_header->find("callerid");
                 if (it2 != out.connection_header->end())
                 {
-                    latched_msgs_.insert({{subscriber->getTopic(), it2->second}, out});
+                    auto const result = latched_msgs_.insert({{subscriber->getTopic(), it2->second}, out});
```

Comment:
```suggestion
                    latched_msgs_[{subscriber->getTopic(), it2->second}] = out;
```

Reply:
The `[]` operator does exactly what's expected here, in case the key doesn't exist in the map, it will be created and when it does exist, it will be updated.

Reply:
Thanks for the suggestion - the `[]` operator is indeed cleaner since it removes the need for the extra manual check (in lines 351-354).

I'll try to update it shortly (I don't have the test environment handy anymore).

Reply:
@AndreasNagel I tried your suggestion and unfortunately it fails to build:
`error: no matching function for call to ‘rosbag::OutgoingMessage::OutgoingMessage()’`

I believe this is due to the behavior of the `[]` operator when a key is not found in the map. From the [docs](https://cplusplus.com/reference/map/map/operator[]/):
> If k does not match the key of any element in the container, the function inserts a new element with that key and returns a reference to its mapped value.

When a key is missing, the `[]` operator tries to create an `OutgoingMessage` object with an empty constructor that does not exist: `OutgoingMessage()`. (The object seems to only have one constructor that requires 4 arguments.)

I didn't find an easy way to go around this without creating an empty constructor for `OutgoingMessage` - I'd prefer to avoid introducing a change like that.